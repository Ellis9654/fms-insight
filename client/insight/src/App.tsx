/* Copyright (c) 2018, John Lenz

All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.

    * Redistributions in binary form must reproduce the above
      copyright notice, this list of conditions and the following
      disclaimer in the documentation and/or other materials provided
      with the distribution.

    * Neither the name of John Lenz, Black Maple Software, SeedTactics,
      nor the names of other contributors may be used to endorse or
      promote products derived from this software without specific
      prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
import * as React from 'react';
import AppBar from 'material-ui/AppBar/AppBar';
import Tabs, { Tab } from 'material-ui/Tabs';
import Typography from 'material-ui/Typography/Typography';
import Toolbar from 'material-ui/Toolbar/Toolbar';
import Hidden from 'material-ui/Hidden/Hidden';
import { Route, Link } from 'react-router-dom';

import Dashboard from './components/Dashboard';
import CostPerPiece from './components/CostPerPiece';
import Efficiency from './components/Efficiency';
import StationMonitor from './components/StationMonitor';
import LoadingIcon from './components/LoadingIcon';
import { requestLastWeek } from './data/events';
import store from './data/store';

// tslint:disable
const LinkTab: any = Tab as any;
// tslint:enable

const tabsStyle = {
  'align-self': 'flex-end',
  'flex-grow': '1'
};

class App extends React.Component {
  componentDidMount() {
    store.dispatch(requestLastWeek());
  }

  render() {
    const tabs = (full: boolean) => (
      <Tabs fullWidth={full} style={full ? {} : tabsStyle} value={location.pathname} onChange={(e, v) => v}>
        <LinkTab label="Dashboard" component={Link} to="/" value="/"/>
        <LinkTab label="Station Monitor" component={Link} to="/station" value="/station"/>
        <LinkTab label="Cost/Piece" component={Link} to="/cost" value="/cost"/>
        <LinkTab label="Efficiency" component={Link} to="/efficiency" value="/efficiency"/>
      </Tabs>
    );

    const largeAppBar = (
      <AppBar position="static">
        <Toolbar>
          <Typography type="title" style={{'margin-right': '2em'}}>Insight</Typography>
          {tabs(false)}
          <LoadingIcon/>
        </Toolbar>
      </AppBar>
    );

    const smallAppBar = (
      <AppBar position="static">
        <Toolbar>
          <Typography type="title">Insight</Typography>
          <div style={{'flex-grow': '1'}}/>
          <LoadingIcon/>
        </Toolbar>
        {tabs(true)}
      </AppBar>
    );

    return (
      <div className="App">
        <Hidden smDown>
          {largeAppBar}
        </Hidden>
        <Hidden mdUp>
          {smallAppBar}
        </Hidden>
        <main style={{margin: '8px'}}>
          <Route exact path="/" component={Dashboard}/>
          <Route exact path="/station" component={StationMonitor}/>
          <Route exact path="/cost" component={CostPerPiece}/>
          <Route exact path="/efficiency" component={Efficiency}/>
        </main>
      </div>
    );
  }
}

export default App;