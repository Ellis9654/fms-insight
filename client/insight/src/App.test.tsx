/* Copyright (c) 2018, John Lenz

All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.

    * Redistributions in binary form must reproduce the above
      copyright notice, this list of conditions and the following
      disclaimer in the documentation and/or other materials provided
      with the distribution.

    * Neither the name of John Lenz, Black Maple Software, SeedTactics,
      nor the names of other contributors may be used to endorse or
      promote products derived from this software without specific
      prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
import * as React from 'react';
import { shallow } from 'enzyme';

import { App, AppProps } from './App';
import * as routes from './data/routes';

function appProps(current: routes.RouteLocation): AppProps {
  let monitor: routes.StationMonitorType = routes.StationMonitorType.LoadUnload;
  switch (current) {
    case routes.RouteLocation.InspectionMonitor:
      monitor = routes.StationMonitorType.Inspection;
      break;
    case routes.RouteLocation.WashMonitor:
      monitor = routes.StationMonitorType.Wash;
  }
  return {
    route: {
      current,
      station_monitor: monitor,
      selected_load_id: 5,
      load_queues: ["q1", "q1"],
      load_free_material: false,
      standalone_queues: ["r1", "r2"],
      standalone_free_material: true,
    },
    setRoute: jest.fn(),
  };
}

it('renders the dashboard', () => {
  const props = appProps(routes.RouteLocation.Dashboard);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("dashboard");
  expect(header).toMatchSnapshot("dashboard header");

  // tslint:disable-next-line:no-any
  const onTabChange = header.find("WithStyles(Tabs)").first().prop("onChange") as any;
  onTabChange(null, routes.RouteLocation.CostPerPiece);
  expect(props.setRoute).toHaveBeenCalledWith({ty: routes.RouteLocation.CostPerPiece, curSt: props.route});
});

it('renders the load monitor', () => {
  const props = appProps(routes.RouteLocation.LoadMonitor);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("station monitor");
  expect(header).toMatchSnapshot("station monitor header");
});

it('renders the inspection monitor', () => {
  const props = appProps(routes.RouteLocation.InspectionMonitor);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("inspection monitor");
  expect(header).toMatchSnapshot("inspection monitor header");
});

it('renders the wash monitor', () => {
  const props = appProps(routes.RouteLocation.WashMonitor);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("wash monitor");
  expect(header).toMatchSnapshot("wash monitor header");
});

it('renders the queue monitor', () => {
  const props = appProps(routes.RouteLocation.Queues);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("queue monitor");
  expect(header).toMatchSnapshot("queue monitor header");
});

it('renders the all material monitor', () => {
  const props = appProps(routes.RouteLocation.AllMaterial);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("all material monitor");
  expect(header).toMatchSnapshot("all material monitor header");
});

it('renders the cost per piece', () => {
  const props = appProps(routes.RouteLocation.CostPerPiece);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("cost per piece");
  expect(header).toMatchSnapshot("cost per piece header");
});

it('renders the efficiency', () => {
  const props = appProps(routes.RouteLocation.Efficiency);
  const val = shallow(<App {...props}/>);
  const header = val.find("Header").dive();
  expect(val).toMatchSnapshot("efficiency");
  expect(header).toMatchSnapshot("efficiency header");
});