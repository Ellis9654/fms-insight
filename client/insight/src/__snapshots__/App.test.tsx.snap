// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the all material monitor: all material monitor 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_AllMaterial",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_LoadUnload",
      }
    }
    setRoute={[MockFunction]}
  />
  <StationMonitor
    monitor_type="StationType_AllMaterial"
  />
</div>
`;

exports[`renders the all material monitor: all material monitor header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={1}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={1}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the cost per piece: cost per piece 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_CostPerPiece",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_LoadUnload",
      }
    }
    setRoute={[MockFunction]}
  />
  <CostPerPiece />
</div>
`;

exports[`renders the cost per piece: cost per piece header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={3}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={3}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the dashboard: dashboard 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_Dashboard",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_LoadUnload",
      }
    }
    setRoute={[MockFunction]}
  />
  <Dashboard />
</div>
`;

exports[`renders the dashboard: dashboard header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={0}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={0}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the efficiency: efficiency 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_Efficiency",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_LoadUnload",
      }
    }
    setRoute={[MockFunction]}
  />
  <Efficiency />
</div>
`;

exports[`renders the efficiency: efficiency header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={2}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={2}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the inspection monitor: inspection monitor 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_Inspection",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_Insp",
      }
    }
    setRoute={[MockFunction]}
  />
  <StationMonitor
    monitor_type="StationType_Insp"
  />
</div>
`;

exports[`renders the inspection monitor: inspection monitor header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={1}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={1}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the load monitor: station monitor 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_LoadMonitor",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_LoadUnload",
      }
    }
    setRoute={[MockFunction]}
  />
  <StationMonitor
    monitor_type="StationType_LoadUnload"
  />
</div>
`;

exports[`renders the load monitor: station monitor header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={1}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={1}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the queue monitor: queue monitor 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_Queues",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_LoadUnload",
      }
    }
    setRoute={[MockFunction]}
  />
  <StationMonitor
    monitor_type="StationType_Queues"
  />
</div>
`;

exports[`renders the queue monitor: queue monitor header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={1}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={1}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;

exports[`renders the wash monitor: wash monitor 1`] = `
<div
  id="App"
>
  <Header
    routeState={
      Object {
        "current": "ROUTE_Wash",
        "load_free_material": false,
        "load_queues": Array [
          "q1",
          "q1",
        ],
        "selected_load_id": 5,
        "standalone_free_material": true,
        "standalone_queues": Array [
          "r1",
          "r2",
        ],
        "station_monitor": "StationType_Wash",
      }
    }
    setRoute={[MockFunction]}
  />
  <StationMonitor
    monitor_type="StationType_Wash"
  />
</div>
`;

exports[`renders the wash monitor: wash monitor header 1`] = `
<nav
  id="navHeader"
>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          style={
            Object {
              "marginRight": "2em",
            }
          }
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <WithStyles(Tabs)
          fullWidth={false}
          onChange={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "flexGrow": 1,
            }
          }
          value={1}
        >
          <WithStyles(Tab)
            label="Dashboard"
            value={0}
          />
          <WithStyles(Tab)
            label="Station Monitor"
            value={1}
          />
          <WithStyles(Tab)
            label="Efficiency"
            value={2}
          />
          <WithStyles(Tab)
            label="Cost/Piece"
            value={3}
          />
        </WithStyles(Tabs)>
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(AppBar)
      position="static"
    >
      <WithStyles(Toolbar)>
        <WithStyles(Typography)
          variant="title"
        >
          Insight
        </WithStyles(Typography)>
        <div
          style={
            Object {
              "flexGrow": 1,
            }
          }
        />
        <Connect(LoadingIcon) />
      </WithStyles(Toolbar)>
      <WithStyles(Tabs)
        fullWidth={true}
        onChange={[Function]}
        style={Object {}}
        value={1}
      >
        <WithStyles(Tab)
          label="Dashboard"
          value={0}
        />
        <WithStyles(Tab)
          label="Station Monitor"
          value={1}
        />
        <WithStyles(Tab)
          label="Efficiency"
          value={2}
        />
        <WithStyles(Tab)
          label="Cost/Piece"
          value={3}
        />
      </WithStyles(Tabs)>
    </WithStyles(AppBar)>
  </Hidden>
</nav>
`;
